src/jitlayers.cpp

```diff
@@ -52,6 +54,9 @@ static void addOptimizationPasses(T *PM)
 #ifndef INSTCOMBINE_BUG
     PM->add(createInstructionCombiningPass()); // Cleanup for scalarrepl.
 #endif
+    // Let the InstCombine pass remove the unnecessary load of
+    // safepoint address first
+    PM->add(createLowerPTLSPass(imaging_mode, tbaa_const));
```

Comment:
this is a module pass, but `addOptimizationPasses` was templated to accept the FunctionPassManager too. I think we only need to make module pass managers now (which might also make the LowerGCFramePass easier, since it can walk the uses list instead of walking the Function Instructions), so it might be good to update the function signature.